src/mcts/search.cc

```diff
@@ -1291,9 +1302,20 @@ void SearchWorker::ExtendNode(Node* node) {
       return;
     }
 
-    if (history_.Last().GetRepetitions() >= 2) {
+    const auto repetitions = history_.Last().GetRepetitions();
+    // Mark two-fold repetitions as draws according to settings.
+    // Depth starts with 1 at root, so number of plies in PV is depth - 1.
+    if (repetitions >= 2) {
       node->MakeTerminal(GameResult::DRAW);
       return;
+    } else if (repetitions == 1 && params_.GetTwoFoldDraws() &&
+               depth - 1 >= 4 &&
```

Comment:
This is probably no longer helping, now that `GetPliesSincePrevRepetition()` is very lightweight.

Reply:
That is of course correct. I think my rationale behind keeping it was that once we try to implement the search as a DAG we have to fall back to something like this, but for one this isn't a reason to keep code confusing, and also I likely won't forget it anyway.
There is no situation where we expect `GetPliesSincePrevRepetition()` to fail here, right?

Reply:
`GetPliesSincePrevRepetition()` is now always consistent with `GetRepetitions()`.

Reply:
Thanks for checking. Marking this as resolved.

Reply:
Was this intentionally reverted?

Reply:
There were 2 points where I checked for `depth - 1 >= 4` before checking the actual condition, here and when making it non terminal after tree reuse. We removed the additional check there because it was 0 cost to do the direct check, but here I can't judge by myself whether calling history is more or less costly than the additional check.

Reply:
In that case, moving it before `params_.GetTwoFoldDraws()` is probably an additional small gain.

Reply:
I agree, especially given that I think `two-fold-draws` should be `true` per default. Changed the order.